gadget/update.go

```diff
@@ -322,16 +353,32 @@ func resolveUpdate(oldVol *PartiallyLaidOutVolume, newVol *LaidOutVolume, policy
 		// assets are assumed to be backwards compatible, once deployed
 		// are not rolled back or replaced unless a higher edition is
 		// available
-		if policy(&oldStruct, &newStruct) {
+		if update, filter := policy(&oldStruct, &newStruct); update {
 			updates = append(updates, updatePair{
 				from: &oldVol.LaidOutStructure[j],
 				to:   &newVol.LaidOutStructure[j],
+
+				resolvedContentFilter: filter,
 			})
 		}
 	}
 	return updates, nil
 }
 
+func filterUpdates(updates []updatePair) {
+	for _, update := range updates {
+		if update.resolvedContentFilter != nil {
```

Comment:
nitpick, given a fair bit of nesting, it's probably easier to follow if we have:
```
for _, update := range updates {
	if update.resolvedContentFilter == nil {
		// this content does not need to be filtered
		continue
	}
	for _, rn := range update.to.ResolvedContent {
			if update.resolvedContentFilter(&rn) {
				filteredResolvedContent = append(filteredResolvedContent, rn)
			}
		}
		update.to.ResolvedContent = filteredResolvedContent
	}
}
```

Reply:
I like this suggestion! I will address it in a followup